### ISSUE TITLE
Missing mime type for Nim-lang

### ISSUE THREAD

**Issue by jeremie2** [ubuntu/yaru#1868]
Hi!
A mime type icon for [Nim](https://nim-lang.org/) language files is missing. I tried to create a text-x-nim.svg file:
![nim](https://user-images.githubusercontent.com/30863448/74078068-277df400-4a26-11ea-8dad-c2a86b2528e6.png)

Is it OK?

**Comment by madsrh** [comment]
Fantastic work @jeremie2 🎉 

Three things:
1. Could you make a PR so we can test this?

2. On my 19.10 default installation there is no support for .nim 😞 Did you install something that added support for this?

![image](https://user-images.githubusercontent.com/3986894/74082147-6b3d2180-4a57-11ea-8a98-f0da0c101ed2.png)


3. I know that we have other icons with a similar "issue", but could you add a shadow? I don't know if this is out of style - @ubuntujaggers will know 😉 
![image](https://user-images.githubusercontent.com/3986894/74082118-02ee4000-4a57-11ea-8cc1-165d57067609.png)

**Comment by meetdilip** [comment]
Other white + logo mime types have a shadow through the middle. Existing icons

![application-vnd microsoft portable-executable](https://user-images.githubusercontent.com/6965443/74083407-b5d79180-4a89-11ea-8f2e-ad1deeb1285f.png)

![application-json](https://user-images.githubusercontent.com/6965443/74083410-b708be80-4a89-11ea-98ca-34a9a752ecf9.png)

It could be just the logos maybe.

**Comment by Feichtmeier** [comment]
This might help:

https://snwh.org/suru/guidelines

**Comment by jeremie2** [comment]
> Fantastic work @jeremie2 tada

Thank you!

> 3. I know that we have other icons with a similar "issue", but could you add a shadow?

I didn't put any shadow because I have seen that all the mime-type icons related to program language files got only a "flat" official logo with no shadow.
So I think this is a guideline.

<br>
That said, Huston we have problem.... I'm a noob :stuck_out_tongue:
It's the first time I'm doing something like that and a few things are actually not clear.<br><br><br>

>  2. On my 19.10 default installation there is no support for .nim disappointed Did you install something that added support for this?

I found a few howtos about adding a new line into `/etc/mime.types`, but didn't work..

Then I found [this tutorial](https://coderwall.com/p/qjda2q/create-new-mime-type-and-assign-an-icon-to-it-in-ubuntu) (custom xml file into `/usr/share/mime/packages`) and it worked.
The problem is that you have to save a *.svg* icon into `/usr/share/icons/Yaru/scalable` and as I can see it isn't the way Yaru works.

Basically I don't know how it works, but I'm almost sure some extra configuration files need to be changed outside Yaru folder, in order to associate a new mime type icon.<br><br>

>  1. Could you make a PR so we can test this?

Sure I would like!
Just one question, shell I create every single _.png_ file into every single *mimetype* folder for each size? Or is it there a system to do that automatically starting from the text-x-nim.svg file?

Sorry if it's something obvious!

Then, all the steps on [Contributing to Yaru](https://github.com/ubuntu/yaru/blob/master/CONTRIBUTING.md) related to git commands seem to be clear :wink:

**Comment by madsrh** [comment]
Nothing is obvious! We're planning to make a short video about icons soon (I did [one about uploading with Git](https://www.youtube.com/watch?v=KZPKJX0eKkY))

You'll have to:
- Download Yaru with `git clone https://github.com/ubuntu/yaru.git`
- Place the template file (contaning all sizes) in the mimetype folder:

![image](https://user-images.githubusercontent.com/3986894/74092647-41bbde80-4ac7-11ea-9690-3212403a77da.png)

Then from the Fullcolor folder render the icon as png with `render-bitmaps.py`. You don't need to type the svg extension. 

![image](https://user-images.githubusercontent.com/3986894/74092701-fd7d0e00-4ac7-11ea-909b-76721d0c831d.png)

Build the theme:
![image](https://user-images.githubusercontent.com/3986894/74092711-256c7180-4ac8-11ea-9438-bad25f75bd01.png)

> Then I found this tutorial

I tried the exact same tutorial, with without any luck. I'll give it another go :wink:

**Comment by meetdilip** [comment]
Use a blog if you don't mind @madsrh . It would be easier to keep track of them.

**Comment by jeremie2** [comment]
Thanks @madsrh 

I tried the commands to create all the _.png_ files starting from _text-x-nim.svg_ but I think something went wrong. Can't see any _text-x-nim.png_ in `~/yaru/icons/Suru` or `/usr/share/icons/Yaru` sub-directories.

This are all the steps:
```
~$ git clone https://github.com/ubuntu/yaru.git
~$ cp text-x-nim.svg yaru/icons/src/fullcolor/mimetypes/
~$ cd yaru/icons/src/fullcolor/
~/yaru/icons/src/fullcolor$ ./render-bitmaps.py applications-apk
```
I think before **ninja install** there should be **meson build** command:
```
~/yaru/icons/src/fullcolor$ cd ~/yaru
~/yaru$ meson build
~/yaru$ cd build
~/yaru/build$ sudo ninja install
```

I didn't receive any error message, maybe there is something wrong with _text-x-nim.svg_ itself.
If anyone would like to check, it's possible to **download** _text-x-nim.svg_ [HERE](https://github.com/jeremie2/icons/tree/master).

<br>

>   Use a blog if you don't mind @madsrh . It would be easier to keep track of them.

Using the [CONTRIBUTING.md](https://github.com/ubuntu/yaru/blob/master/CONTRIBUTING.md) file or the [yaru/wiki](https://github.com/ubuntu/yaru/wiki) would be even better 🙂
(P.S. of course videos are welcome too!!)

**Comment by ubuntujaggers** [comment]
Currently on a train but will help with this PR tonight 👍

**Comment by ubuntujaggers** [comment]
Aha  :)

The script that exports the .pngs takes the icon name from the baseplate layer.

Layers menu --> Layers... --> Click the eye next to the layer called "Baseplate" to make it visible:

![image](https://user-images.githubusercontent.com/38893390/74108168-610e4680-4b6f-11ea-9cf7-d2e625ae3887.png)

Using the text edit tool (button looks like "A|") change **text-x-python** to **text-x-nim**.  Then make the Baseplate layer invisible again.  Save the svg and re-run the script...  does it work now?

**Comment by ubuntujaggers** [comment]
And yep, I'll make a vid early in the week - what app did you use again, @madsrh ?

**Comment by madsrh** [comment]
> what app did you use again, madsrh ?

[Kazam](https://launchpad.net/kazam)

> I'll make a vid early in the week

🎉 💪  Would it be okay if I made a short video too?

**Comment by jeremie2** [comment]
> Using the text edit tool (button looks like "A|") change text-x-python to text-x-nim. Then make the Baseplate layer invisible again. Save the svg and re-run the script... does it work now?

Thanks @ubuntujaggers,
Sorry, I've modified the svg and run all the steps again, but still doesn't work.

Could it be the `./render-bitmaps.py applications-apk` command? Maybe something else instead of **applications-apk**?

**Comment by ubuntujaggers** [comment]
It should be the filename at the end of the command minus the .SVG... try ./render-bitmaps.py text-x-nim.svg ☺️

**Comment by ubuntujaggers** [comment]
Argh, sorry, got it wrong:

python render-bitmaps.py text-x-nim

...and if it doesn't work try sudo apt-get install python?

**Comment by madsrh** [comment]
I just uploaded this short video:

https://www.youtube.com/watch?v=_fGr2Lt6W8c

**Comment by jeremie2** [comment]
@ubuntujaggers it works!!!! Thankyouverymuchindeed 😀

**./render-bitmaps.py text-x-nim** command is fine as _render-bitmaps.py_ begins with `#!/usr/bin/python3`

I'm ready for next step.. hope the pull request will run smoothly!

**Comment by jeremie2** [comment]
Sorry! Forgot to mention I've open PR #1888

Don't know if this issue has to be closed now, anyway thanks to everyone for your big help!!
Great community 😉

**Comment by madsrh** [comment]
We close the issue when PR is merged

> Great community 😉

Indeed 😄
